openedx/core/djangoapps/content/block_structure/models.py

```diff
@@ -234,11 +234,11 @@ def update_or_create(cls, serialized_data, data_usage_key, **kwargs):
             with _storage_error_handling(bs_model, operation):
                 bs_model.data.save('', ContentFile(serialized_data))
 
-        cls._log(bs_model, operation, serialized_data)
-
         if not created:
             cls._prune_files(data_usage_key)
-
+            
+        cls._log(bs_model, operation, serialized_data)
```

Comment:
@Yagnesh1998 I'm worried about moving the log message lower.  It would mean if there is a failure in pruning, we will not get the log message about the document creation or update.  That is confusing and might make it harder to debug the issue.  However, looking at the atomic transaction, it looks like there is plenty of logging in the error handler there so I think this is fine to do and won't harm anything.